pyartcd/pyartcd/pipelines/gen_assembly.py

```diff
@@ -0,0 +1,233 @@
+            await self._slack_client.say(message, slack_thread)
+        except Exception as err:
+            error_message = f"Error generating assembly definition: {err}\n {traceback.format_exc()}"
+            self._logger.error(error_message)
+            await self._slack_client.say(error_message, slack_thread)
+            raise
+
+    async def _get_nightlies(self):
+        """ Get nightlies from Release Controllers
+        :param release: release field for rebase
+        :return: NVRs
+        """
+        cmd = [
+            "doozer",
+            "--group", self.group,
+            "--assembly", "stream",
+        ]
+        if self.arches:
+            cmd.append("--arches")
+            cmd.append(",".join(self.arches))
+        cmd.append("get-nightlies")
+        if self.allow_pending:
+            cmd.append("--allow-pending")
+        if self.allow_rejected:
+            cmd.append("--allow-rejected")
+        if self.allow_inconsistency:
+            cmd.append("--allow-inconsistency")
+        for nightly in self.nightlies:
+            cmd.append(f"--matching={nightly}")
+
+        _, out, _ = await exectools.cmd_gather_async(cmd, stderr=None, env=self._doozer_env_vars)
+        return out.strip().split()
+
+    async def _gen_assembly_from_releases(self, candidate_nightlies: Iterable[str]) -> OrderedDict:
+        """ Run doozer release:gen-assembly from-releases
+        :return: Assembly definition
+        """
+        cmd = [
+            "doozer",
+            "--group", self.group,
+            "--assembly", "stream",
+        ]
+        if self.arches:
+            cmd.append("--arches")
+            cmd.append(",".join(self.arches))
+        cmd.append("release:gen-assembly")
+        cmd.append(f"--name={self.assembly}")
+        cmd.append("from-releases")
+        for nightly in candidate_nightlies:
+            cmd.append(f"--nightly={nightly}")
+        if self.custom:
+            cmd.append("--custom")
+        else:
+            if self.in_flight:
+                cmd.append(f"--in-flight={self.in_flight}")
+            for previous in self.previous_list:
+                cmd.append(f"--previous={previous}")
+            if self.auto_previous:
+                cmd.append("--auto-previous")
+        _, out, _ = await exectools.cmd_gather_async(cmd, stderr=None, env=self._doozer_env_vars)
+        return yaml.load(out)
+
+    async def _create_or_update_pull_request(self, fn: Callable[[OrderedDict], OrderedDict]):
+        """ Create or update pull request for ocp-build-data
+        :param fn: a callback function that makes changes to releases.yml
+        """
+        self._logger.info("Creating ocp-build-data PR...")
+        # Clone ocp-build-data
+        build_data_path = self._working_dir / "ocp-build-data-push"
+        build_data = GitRepository(build_data_path, dry_run=self.runtime.dry_run)
+        ocp_build_data_repo_push_url = self.runtime.config["build_config"]["ocp_build_data_repo_push_url"]
```

Comment:
The repo to push is actually configured at https://github.com/openshift/aos-cd-jobs/blob/94905e9c5c1a7e381cdda8b9a85cb0fea7f5254e/config/artcd.toml#L8. It can be changed to the fork, but we need some tests (and maybe also need to pre-create the fork) before we actually make that change.

Currently those auto created branches always start with auto- . We can clean them up easily in the future.